multiplexing_server.py

```diff
@@ -0,0 +1,175 @@
+import socket
+import time
+import select
+
+# global variables
+MAX_MSG_LENGTH = 1024
+SERVER_PORT = 5555
+SERVER_IP = "0.0.0.0"
+MANAGERS_ARRAY = ["@ariel", "@shua"]
+muted_socket_array = []
+
+
+# splits the message to get the info
+def parse_message(message_to_parse):
+    # getting name
+    index = 0
+    num_str = ""
+    while message_to_parse[index].isnumeric():
+        num_str += message_to_parse[index]
+        index += 1
+
+    username_length = int(num_str)
+    username = message_to_parse[index:index + username_length]
+    index += username_length
+    command = message_to_parse[index]
+    index += 1
+    if command == "5":
+        num_str = ""
+        while message_to_parse[index].isnumeric():
+            num_str += message_to_parse[index]
+            index += 1
+        name_for_send_length = int(num_str)
+        name_for_send = message_to_parse[index: index + name_for_send_length]
+        num_str = ""
+        while message_to_parse[index].isnumeric():
+            num_str += message_to_parse[index]
+            index += 1
+        msg_to_send_length = int(num_str)
+        msg_to_send = message_to_parse[index: index + msg_to_send_length]
+        return username, command, msg_to_send, name_for_send
+    num_str = ""
+    while message_to_parse[index].isnumeric():
+        num_str += message_to_parse[index]
+        index += 1
+
+    data_length = int(num_str)
+    data = message_to_parse[index: index + data_length]
+    print(data)
+    return username, command, data, ""
+
+
+# an action to kick members
+def kick_action(selected_name):
+    selected_socket = dict_of_sockets_name[selected_name]
+    print(dict_of_sockets_name[selected_name])
+    open_client_sockets.remove(selected_socket)
+    selected_socket.close()
+    return f"{selected_name} has been kicked from the chat!"
+
+def mute_action(selected_name):
+    print(selected_name)
+    selected_socket = dict_of_sockets_name[selected_name]
+    muted_socket_array.append(selected_socket)
+    return "you cannot speak here"
+
+def appointment_of_manager(selected_name):
+    manger_selected_name = "@" + selected_name
+    MANAGERS_ARRAY.append(selected_name)
+    dict_of_sockets_name[manger_selected_name] = dict_of_sockets_name.pop(selected_name)
```

Comment:
why do you need to pop from `dict_of_sockets_name`
please keep a dictionary that looks like this:
```py
users = {
  "SOME_NAME_HERE": {
     "socket": socket_object,
     "isAdmin": false # or true
  }
}
```